test/end-to-end.js

```diff
@@ -970,6 +976,44 @@ describe('end-to-end', function() {
       });
     });
 
+    it('shows error behavior responses in development mode', function(done) {
+      var app = loopback({ localRegistry: true, loadBuiltinModels: true });
+      var bootOptions = {
+        appRootDir: SANDBOX,
+        env: 'development'
+      };
+      boot(app, bootOptions, function(err) {
+        if (err) return done(err);
+        app.listen(0, function() {
+          request(app)
+            .get('/url-does-not-exist')
+            .end(function(err, res) {
+              expect(res.error.text.match(/url-does-not-exist/));
+              done(err);
+          });
+        });
+      });
+    });
+
+    it('shows error behavior responses in production mode', function(done) {
+      var app = loopback({ localRegistry: true, loadBuiltinModels: true });
+      var bootOptions = {
+        appRootDir: SANDBOX,
+        env: 'production'
+      };
+      boot(app, bootOptions, function(err) {
+        if (err) return done(err);
+        app.listen(0, function() {
+          request(app)
+            .get('/url-does-not-exist')
+            .end(function(err, res) {
+              expect(res.error.text.match(/url-does-not-exist/));
```

Comment:
In production mode, we need to check that the error response does not contain the stack trace. To make the assertion future-proof, we should not rely on particular (deep) property name like `error.stack`, but scan the string response body instead.

Also handle `err` from supertest as mentioned in https://github.com/strongloop/loopback-workspace/pull/285/files#r66961915